web/src/main/java/com/graphhopper/http/GHBaseServlet.java

```diff
@@ -65,33 +71,33 @@ protected void writeJson(HttpServletRequest req, HttpServletResponse res, JSONOb
             }
 
             if (debug)
-                writeResponse(res, callbackName + "(" + json.toString(2) + ")");
+                writeResponse(res, callbackName + "(" + objectMapper.writeValueAsString(json) /*.toString(2)*/ + ")");
             else
-                writeResponse(res, callbackName + "(" + json.toString(0) + ")");
+                writeResponse(res, callbackName + "(" + objectMapper.writeValueAsString(json) /*.toString(0)*/ + ")");
 
         } else {
             res.setContentType("application/json");
             if (debug)
-                writeResponse(res, json.toString(2));
+                writeResponse(res, objectMapper.writeValueAsString(json) /*.toString(2)*/);
```

Comment:
The super-advanced DI-way would probably be to bind() a request-scoped ``ObjectWriter`` which is debug or non-debug, and then inject that configured ``ObjectWriter`` everywhere..